src/MQTTAsyncUtils.c

```diff
@@ -2584,9 +2620,9 @@ void Protocol_processPublication(Publish* publish, Clients* client, int allocate
 		if (client->persistence)
 			MQTTPersistence_persistQueueEntry(client, (MQTTPersistence_qEntry*)qe);
 #endif
+		publish->topic = NULL; /* Prevent topic name free as reference was passed to qe */
 	}
 exit:
-	publish->topic = NULL;
```

Comment:
Moving this line inside the condition above fixes an active leak of the topic name.